Source/NSFontCollection.m

```diff
@@ -0,0 +1,341 @@
+      id name = nil;
+
+      en = [names objectEnumerator];
+      while ((name = [en nextObject]) != nil)
+        {
+          NSFont *font = [NSFont fontWithName: name size: 0.0]; // get default size
+          [_fonts addObject: font];
+        }
+    }
+}
+
++ (NSFontCollection *) fontCollectionWithDescriptors: (NSArray *)queryDescriptors
+{
+  NSFontCollection *fc = [[NSFontCollection alloc] init];
+  ASSIGNCOPY(fc->_queryDescriptors, queryDescriptors);
+  return fc;
+}
+
++ (NSFontCollection *) fontCollectionWithAllAvailableDescriptors
+{
+  return [self fontCollectionWithDescriptors:
+                 [[GSFontEnumerator sharedEnumerator] availableFontDescriptors]];
+}
+
++ (NSFontCollection *) fontCollectionWithLocale: (NSLocale *)locale
+{
+  return [self fontCollectionWithAllAvailableDescriptors];
+}
+
++ (BOOL) showFontCollection: (NSFontCollection *)collection
+                   withName: (NSFontCollectionName)name
+                 visibility: (NSFontCollectionVisibility)visibility
+                      error: (NSError **)error
+{
+  NSNumber *v = [NSNumber numberWithInt: visibility];
+  [__sharedFontCollections setObject: collection
+                             forKey: name];
+  [__sharedFontCollectionsVisibility setObject: v
+                                        forKey: name];
+  return YES;
+}
+
++ (BOOL) hideFontCollectionWithName: (NSFontCollectionName)name
+                         visibility: (NSFontCollectionVisibility)visibility
+                              error: (NSError **)error
+{
+  
+  [__sharedFontCollectionsHidden addObject: name];
+  return YES;
+}
+
++ (BOOL) renameFontCollectionWithName: (NSFontCollectionName)aname
+                           visibility: (NSFontCollectionVisibility)visibility
+                               toName: (NSFontCollectionName)name
+                                error: (NSError **)error
+{
+  NSFontCollection *fc = [__sharedFontCollections objectForKey: aname];
+  [__sharedFontCollections setObject: fc forKey: name];
+  return YES;
+}
+
++ (NSArray *) allFontCollectionNames
+{
+  return [__sharedFontCollections allKeys];
+}
+
++ (NSFontCollection *) fontCollectionWithName: (NSFontCollectionName)name
+{
+  return [__sharedFontCollections objectForKey: name];
+}
+
++ (NSFontCollection *) fontCollectionWithName: (NSFontCollectionName)name
+                                   visibility: (NSFontCollectionVisibility)visibility
+{
+  return [__sharedFontCollections objectForKey: name];
+}
+
+// Descriptors
+- (NSArray *) queryDescriptors  // copy
+{
+  return [_queryDescriptors copy];
+}
+
+- (NSArray *) exclusionDescriptors
+{
+  return [_exclusionDescriptors copy];
+}
+
+- (NSArray *) matchingDescriptors
+{
+  return [self matchingDescriptorsWithOptions: nil];
+}
+
+- (NSArray *) matchingDescriptorsWithOptions: (NSDictionary *)options
+{
+  GSFontEnumerator *fen = [GSFontEnumerator sharedEnumerator];
+  return [fen matchingFontDescriptorsFor: options];
```

Comment:
Here we need to add _queryDescriptors and _exclusionDescriptors to get the actual font descriptors. The options provided here cannot the handled by GSFontEnumerator at the moment. Maybe we should handle them here? As matchingDescriptorsForFamily:options: seems to be the most general method, all others should call that and we need a better implementation there.